src/Serializer/AbstractItemNormalizer.php

```diff
@@ -488,4 +498,51 @@ protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relate
 
         return $iri;
     }
+
+    /**
+     * Sets the object to populate if allowed to and not set already. If you pass identifiers, it will search for them,
+     * otherwise it falls back to the provided IdentifiersExtractor.
+     *
+     * @param mixed  $data
+     * @param string $class
+     * @param array  $context
+     */
+    protected function setObjectToPopulate($data, string $class, array &$context)
+    {
+        if (isset($context[self::OBJECT_TO_POPULATE]) || !\is_array($data)) {
+            return;
+        }
+
+        $updateAllowed = $context['api_allow_update'] ?? false;
+        $identifiers = $this->getIdentifiersForDenormalization($class);
+        $identifiersData = \array_intersect_key($data, array_flip($identifiers));
+
+        if (0 === \count($identifiersData)) {
+            return;
+        }
+
+        if (!$updateAllowed) {
+            throw new InvalidArgumentException('Update is not allowed for this operation.');
+        }
+
+        // TODO: use $this->iriConverter->getIriFromPlainIdentifier() once https://github.com/api-platform/core/pull/1837 is merged.
+        $iri = implode(';', $identifiersData);
+
+        $context[self::OBJECT_TO_POPULATE] = $this->iriConverter->getItemFromIri($iri, $context + ['fetch_data' => true]);
+    }
+
+    /**
+     * Returns the identifiers used when denormalizing an object.
+     * Usually this is just what the IdentifiersExtractor provides but you might want to override this method in your
+     * normalizer depending on the format used. Note: They also have to be writeable to serve as proper identifier.
+     */
+    protected function getIdentifiersForDenormalization(string $class): array
+    {
+        // Backwards compatibility
+        if (null === $this->identifiersExtractor) {
```

Comment:
```php
if (!$this->identifiersExtractor) {
```

Reply:
again, happy to follow style of the project, but `if (null === $variable)` is done quite a lot in this project, so i'm not sure there's any point unless consistency is introduced.